Mage.Sets/src/mage/cards/n/NaturalBalance.java

```diff
@@ -0,0 +1,127 @@
+ *  or implied, of BetaSteward_at_googlemail.com.
+ */
+package mage.cards.n;
+
+import java.util.UUID;
+import mage.abilities.Ability;
+import mage.abilities.dynamicvalue.common.PermanentsOnBattlefieldCount;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.cards.CardsImpl;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.TargetController;
+import mage.constants.Zone;
+import mage.filter.common.FilterBasicLandCard;
+import mage.filter.common.FilterControlledLandPermanent;
+import mage.filter.common.FilterLandPermanent;
+import mage.filter.predicate.permanent.ControllerPredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.players.PlayerList;
+import mage.target.common.TargetCardInLibrary;
+import mage.target.common.TargetControlledPermanent;
+
+/**
+ *
+ * @author Rene - bugisemail at gmail dot com
+ */
+public class NaturalBalance extends CardImpl {
+
+    public NaturalBalance(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{2}{G}{G}");
+
+        // Each player who controls six or more lands chooses five lands he or she controls and sacrifices the rest. Each player who controls four or fewer lands may search his or her library for up to X basic land cards and put them onto the battlefield, where X is five minus the number of lands he or she controls. Then each player who searched his or her library this way shuffles it.
+        this.getSpellAbility().addEffect(new NaturalBalanceEffect());
+    }
+
+    public NaturalBalance(final NaturalBalance card) {
+        super(card);
+    }
+
+    @Override
+    public NaturalBalance copy() {
+        return new NaturalBalance(this);
+    }
+
+    class NaturalBalanceEffect extends OneShotEffect {
+
+        public NaturalBalanceEffect() {
+            super(Outcome.PutCardInPlay);
+            this.staticText = "Each player who controls six or more lands chooses five lands he or she controls and sacrifices the rest. Each player who controls four or fewer lands may search his or her library for up to X basic land cards and put them onto the battlefield, where X is five minus the number of lands he or she controls. Then each player who searched his or her library this way shuffles it.";
+        }
+
+        public NaturalBalanceEffect(final NaturalBalanceEffect effect) {
+            super(effect);
+        }
+
+        @Override
+        public NaturalBalanceEffect copy() {
+            return new NaturalBalanceEffect(this);
+        }
+
+        @Override
+        public boolean apply(Game game, Ability source) {
+            Player controller = game.getPlayer(source.getControllerId());
+            if (controller != null) {
+                PlayerList players = game.getState().getPlayerList(controller.getId());

```

Comment:
Use getPlayersInRange to get only the affected players and not all players at the table.